bundle/src/main/java/com/adobe/acs/commons/util/OverridePathSlingRequestWrapper.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.util;
+
+import com.adobe.cq.sightly.WCMBindings;
+import com.day.text.Text;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.adapter.AdapterManager;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.scripting.SlingBindings;
+import org.apache.sling.api.wrappers.SlingHttpServletRequestWrapper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Request wrapper to get the Resource and SlingBindings objects for a request scoped to a specified path.
+ *
+ * Among other things, this enables injection of sling models from a SlingHttpServletRequest
+ * object rather than simply a Resource, which is required by some sling models for full
+ * functionality.
+ */
+public class OverridePathSlingRequestWrapper extends SlingHttpServletRequestWrapper {
+    private static final String ATTR_SLING_BINDINGS = SlingBindings.class.getName();
+
+    private final SlingBindings myBindings = new SlingBindings();
+    private final Resource resource;
+
+    private final AdapterManager adapterManager;
+    private final Map<Class<?>, Object> adaptersCache = new HashMap<>();
+
+    /**
+     * Constructor.
+     *
+     * @param wrappedRequest SlingHttpServletRequest
+     * @param relPath        String
+     */
+    public OverridePathSlingRequestWrapper(final SlingHttpServletRequest wrappedRequest, final String relPath) {
+        super(wrappedRequest);
+
+        SlingBindings slingBindings = (SlingBindings) getSlingRequest().getAttribute(ATTR_SLING_BINDINGS);
+
+        this.adapterManager = slingBindings.getSling().getService(AdapterManager.class);
+        this.resource = getSlingRequest().getResourceResolver().resolve(getSlingRequest(),
+                Text.fullFilePath(getSlingRequest().getRequestPathInfo().getResourcePath(), relPath));
+        this.myBindings.putAll(slingBindings);
+        this.myBindings.put(WCMBindings.PROPERTIES, this.resource.getValueMap());
+        this.myBindings.put(SlingBindings.RESOURCE, this.resource);
+        this.myBindings.put(SlingBindings.REQUEST, this);
+    }
+
+    @Override
+    public Object getAttribute(final String name) {
+        if (ATTR_SLING_BINDINGS.equals(name)) {
+            return this.myBindings;
+        } else {
+            return super.getAttribute(name);
+        }
+    }
+
+    @Override
+    public Resource getResource() {
+        return this.resource;
+    }
+
+    /**
+     * Overriding `adaptTo` to avoid using the original request as the adaptable.
+     */
+    @Override
+    public <AdapterType> AdapterType adaptTo(Class<AdapterType> type) {
```

Comment:
Per docs when null is returned, a message should be logged.

https://sling.apache.org/apidocs/sling9/org/apache/sling/api/adapter/AdapterFactory.html#getAdapter-java.lang.Object-java.lang.Class-

Reply:
I believe this would actually be the responsibility of the `AdapterFactory` implementation itself, since only it would have the info as to why the adaptation failed.  This function is simply delegating to the adapter factory implementations, similar to `SlingAdaptable` which also does not log a `null` return from `adaptTo`.

Reply:
Aren’t you returning null here if adaptermanger is null? So it seems it should log.warn(“unable to adapt x to y because AdapterManager is null”); ?
Of course if that’s null that’s a different, worse problem, but this should log or per docs?

Reply:
I guess its an edge case possibility, so I covered it now.